src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.Sort.cs

```diff
@@ -14,20 +14,53 @@ public partial class PrimitiveDataFrameColumn<T> : DataFrameColumn
     {
         public new PrimitiveDataFrameColumn<T> Sort(bool ascending = true)
         {
-            PrimitiveDataFrameColumn<long> sortIndices = GetAscendingSortIndices();
-            return Clone(sortIndices, !ascending, NullCount);
+            PrimitiveDataFrameColumn<long> sortIndices;
+            if (ascending)
+            {
+                sortIndices = GetAscendingSortIndices();
+            }
+            else
+            {
+                sortIndices = GetDescendingSortIndices();
+            }
+            return Clone(sortIndices);
         }
 
         internal override PrimitiveDataFrameColumn<long> GetAscendingSortIndices()
         {
             // The return sortIndices contains only the non null indices. 
-            GetSortIndices(Comparer<T>.Default, out PrimitiveDataFrameColumn<long> sortIndices);
+            PrimitiveDataFrameColumn<long> sortIndices = GetSortIndices(Comparer<T>.Default, out PrimitiveDataFrameColumn<long> columnNullIndices);
+            if(columnNullIndices.Length > 0)
```

Comment:
(nit) You don't need a Length check before the `for` loop. The `for` loop will no-op if `Length == 0`.